README.md

```diff
@@ -1,7 +1,190 @@
+```
+
+Note: The standard interpretation of the logical symbols - "∨", "∧", "→", "¬", "∀", "∃" - is assumed throughout. 
+Other symbols: ⊔ ⊓ ⊧ ⊭ ⊦ ⊬ ⊏ ⊐ ⊑ ⊒ ≡
+
+
+1. Provide the truth tables for each of the following propositional logic formulas. State whether each is a tautology, a contradiction, or contingent:
+  ```
+  (a) (¬A→B)∨((A∧¬C)→B) 
+Tautology (see image)
+  (b) (A→B)∧(A→¬B)
+Contingent (see image)
+  (c) (A→(B∨C))∨(C→¬A) 
+Tautology (see image)
+  (d) ((A→B)∧C)∨(A∧D) 
+Contingent (see image)
+  ```
+	
+2. A _literal_ is an atomic formula or the negation of an atomic formula. We say a formula is in _conjunctive normal form_ (CNF) if it is the conjunction of the disjunction of literals. Find propositional logic formulas in CNF equivalent to each of the following:
+  ```
+  (a) (A→B)→C
+  ¬(¬A∨B)∨C
+  (A∧¬B)∨C
+  (A∨C)∧(¬B∨C)
+  (b) (A→(B∨C))∨(C→¬A)
+  (¬A∨(B∨C))∨(¬C∨¬A)
+  ¬A∨B
+  (c) (¬A∧¬B∧C)∨(¬A∧¬C)∨(B∧C)∨A 
+  (¬A∧¬B∧C)∨(¬A∧¬C)∨(B∧C)∨A 
+  Not exactly sure how to apply the distributive rule here. 
+  ```
+  
+3. Let V be the vocabulary of first-order logic consisting of a binary relation P and a unary relation F. Interpret P(x,y) as “x is a parent of y” and F(x) as “x is female.” Where possible define the following formulas in this vocabulary; where not possible, explain why: 
+  
+  ```
+  (a)  B(x,y) that says that x is a brother of y  
+  B(x,y) := ∃z.(P(zx) ∧ P(zy)) ∧ ¬F(x)
+  (b)  A(x,y) that says that x is an aunt of y  
+  A(x,y) := ∃z.P(zy) ∧ ∃w.(P(wz) ∧ P(wx)) ∧ F(x)
+  (c)  C(x,y) that says that x and y are cousins   
+  C(x,y) := ∃z.P(zy) ∧ ∃w.P(wx) ∧ ∃u.(P(uz) ∧ P(uw))
+  (d)  O(x) that says that x is an only child  
+  O(x) := ∃y.P(yx) ∧ ¬∃z.(∃w.P(wz) ∧ (w=y))
+  (e)  T(x) that says that x has exactly two brothers 
+  T(x) := ∃y∃z∃w.(P(yx) ∧ P(yz) ∧ P(yw)) ∧ (P(yq) → ((q=x) ∨ (q=z) ∨ (q=w)))
+
+  ```
+
+4. Let V be a vocabulary of the attribute (concept) language with complements (ALC) consisting of a role name "parent_of" and a concept name "Male". Interpret parent_of as "x is a parent of y" and M as "x is male". Where possible define the following formulas in this vocabulary; where not possible, explain why: 
+  ```
+  ⊔ ⊓ ⊧ ⊭ ⊦ ⊬ ⊏ ⊐ ⊑ ⊒ ≡ ¬
+  parent_of
+  m
+  To be honest I do not know how to solve most of these without introducing some concept like "person". 
+  We definitely want to put some condition after the existential restriction, but I am not able to specify such restriction with the given concepts.
+
+
+  (a)  B that says that x is a brother of y
+  B ≡ m ⊓ ∃parentof.(and some conditions saying that the parentof-filler is the subject of the definition)

```

Comment:
Sent you comments in your project about this

Reply:
> Sent you comments in your project about this

@giacomodecolle This is Ali's proposed definition of Aunt (x is an aunt of y), but the similar is for you lol:
A := ¬M ⊓ (∃p2¯.((parent_of. ≥ 2).(parent_of.person)) ⊔ ¬parent_of.person)

In my view, the best way of writing this formula seems to be as follows:

A ≡ ¬M ⊓ (∃p2¯.(≥ 2 parent_of. (∃parent_of. Person)) ⊔ ¬∃parent_of. Person)

The reason is, when we try to use both Concept (Name) and Role (Name) in a DL-formula, it is either (≥ n r. C) or (≤ n r. C) or (∃r. C) or (∀r. C). That is to mean, a quantifier must be placed immediately before a role name unless the role name is restricted by a numerical. Also, ":=" seems not to be a symbol in DL, but "≡" is.

I think similar syntactic changes may also be applied to other definitions in Question 4.

Ok, the above is a comment on syntax. But another worry is about semantics. This definition of Aunt seems to say: A guy x is an Aunt iff:
(i) x is female, **and** 
(ii) [**(a)** x is one of at least two kids produced by a parent w, who (w) has at least two kids _each of who_ has a kid; **or** **(b)** x has no kid.]

When we say in (a) that w has at least two kids, w may have 5 kids! In that possible case, (a) implies that each of those 5 kids of w has a kid. But this seems too strong because it is enough that only one of those 5 kids (say z) has a kid who happens to be y. Only adding (b) does not exercise a reasonable limit here.